src/main/java/uk/gov/justice/digital/hmpps/cmd/api/service/ShiftService.kt

```diff
@@ -77,11 +77,12 @@ class ShiftService(
         if (
           (activity == FullDayActivityType.REST_DAY && onlyHasBreaksOrNightShiftFinish(shiftTasks, FullDayActivityType.REST_DAY)) ||
           (activity == FullDayActivityType.HOLIDAY && onlyHasBreaksOrNightShiftFinish(shiftTasks, FullDayActivityType.HOLIDAY)) ||
-          (activity == FullDayActivityType.ILLNESS) ||
+          (activity == FullDayActivityType.ILLNESS && onlyHasBreaksOrNightShiftFinish(shiftTasks, FullDayActivityType.ILLNESS)) ||
           (activity == FullDayActivityType.ABSENCE && onlyHasBreaksOrNightShiftFinish(shiftTasks, FullDayActivityType.ABSENCE)) ||
           (activity == FullDayActivityType.TU_OFFICIALS_LEAVE_DAYS) ||
           (activity == FullDayActivityType.TU_OFFICIALS_LEAVE_HOURS) ||
           (activity == FullDayActivityType.TOIL && isStartType(it.displayType)) ||
+          (activity == FullDayActivityType.TOIL && onlyHasBreaksOrNightShiftFinish(shiftTasks, FullDayActivityType.TOIL)) ||
```

Comment:
are you sure you want 2 TOIL rows?

Reply:
When TOIL is part of other activities in the shift then there will be a break and an end time during the day i.e not night shift finish, so I'm thinking that both cases need to be catered for, but will know for sure soon enough!